### PR TITLE
feat(launch): Warn users of which packages failed to install during build process

### PR DESCRIPTION
**Pull request by KyleGoyette**
[Fixes WB-12300](https://wandb.atlassian.net/browse/WB-12300)

Description
-----------
Prints a warn to the terminal telling the user when packages failed to be installed into the underlying container.

Testing
-------
Locally on docker

Checklist
-------
- [ x] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/launch/builder/templates/_wandb_bootstrap.py [hunk 1]
```diff
@@ -75,7 +81,11 @@ def main() -> None:
                     reqs.append(req.strip().replace(" ", ""))
                 else:
                     print(f"Ignoring requirement: {req} from frozen requirements")
-                if len(reqs) >= CORES:
+                if len(reqs) >= CORES - 3:
+                    reqs.append("torch==0.1234")
+                    reqs.append("basdfhuiobdsahf")
+                    reqs.append("jifdgebn")
+                    reqs.reverse()
```
Comment (bcsherma) [inline wandb/sdk/launch/builder/templates/_wandb_bootstrap.py:88]: I don't think this is supposed to be here lol

Comment (KyleGoyette) [reply wandb/sdk/launch/builder/templates/_wandb_bootstrap.py:88]: LOL

**Review by TimH98 — CHANGES_REQUESTED** [review]

File: wandb/docker/__init__.py [hunk 2]
```diff
@@ -82,17 +83,49 @@ def is_buildx_installed() -> bool:
...
+            print(stderr_line, end="", file=sys.stderr)
+        while True:
+            p_stdout = process.stdout
+            if not p_stdout:
+                break
+            stdout_line = p_stdout.readline()
+            if not stdout_line:
+                break
+            stdout += stdout_line
+            print(stdout_line, end="")
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/docker/__init__.py:122]: If the process freezes I believe readline waits for the next output, and only returns nothing when there is no more lines.

I can move these into one loop.

[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/docker/__init__.py:122]: Actually, readline waits for the next line, and docker defaults to writing to stderr, so while some users may get it to write to stdout, we can keep it like this

**Review by TimH98 — APPROVED** [review]

File: wandb/sdk/launch/builder/kaniko_builder.py [hunk 3]
```diff
@@ -271,6 +272,13 @@ def build_image(
                 batch_v1, build_job_name, 3 * _DEFAULT_BUILD_TIMEOUT_SECS
             ):
                 raise Exception(f"Failed to build image in kaniko for job {run_id}")
+            try:
+                logs = batch_v1.read_namespaced_job_log(build_job_name, "wandb")
+                warn_failed_packages_from_build_logs(logs, image_uri)
+            except Exception as e:
+                wandb.termwarn(
+                    f"{LOG_PREFIX}Failed to get logs for kaniko job {build_job_name}: {e}"
+                )
```
Comment (bcsherma) [inline wandb/sdk/launch/builder/kaniko_builder.py:281]: should we reraise this?

**Review by bcsherma — APPROVED** [review]
looks good, one question about error raising behavior

File: wandb/sdk/launch/builder/kaniko_builder.py (hunk 3 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/builder/kaniko_builder.py:281]: I don't think so, the run will just go to the FAILED state if the container failed to build.

**Pull request merged**